source/unified-test-format/unified-test-format.rst

```diff
@@ -946,6 +1011,92 @@ The structure of this object is as follows:
+
+.. _expectedEvent_poolReadyEvent:
+
+- ``poolReadyEvent``: Optional object. If present, this object MUST be an
+  empty document as all fields in this event are non-deterministic.
+
+.. _expectedEvent_poolClearedEvent:
+
+- ``poolClearedEvent``: Optional object. Assertions for one or more
+  `PoolClearedEvent <../connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#events>`__
+  fields.
+
+  The structure of this object is as follows:
+
+  - ``serverId``: Defined in `hasServerId`_.
+
+.. _expectedEvent_poolClosedEvent:
+
+- ``poolClosedEvent``: Optional object. If present, this object MUST be an
+  empty document as all fields in this event are non-deterministic.
+
+.. _expectedEvent_connectionCreatedEvent:
+
+- ``connectionCreatedEvent``: Optional object. If present, this object MUST be
+  an empty document as all fields in this event are non-deterministic.
+
+.. _expectedEvent_connectionReadyEvent:
+
+- ``connectionReadyEvent``: Optional object. If present, this object MUST be an
+  empty document as all fields in this event are non-deterministic.
+
+.. _expectedEvent_connectionClosedEvent:
+
+- ``connectionClosedEvent``: Optional object. Assertions for one or more
+  `ConnectionClosedEvent <../connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#events>`__
+  fields.
+
+  The structure of this object is as follows:
+
+  - ``reason``: Optional string. Test runners MUST assert that the reason in the
+    published event matches this value. Valid values for this field are defined
+    in the CMAP spec.
+
+.. _expectedEvent_connectionCheckOutStartedEvent:
+
+- ``connectionCheckOutStartedEvent``: Optional object. If present, this object
+  MUST be an empty document as all fields in this event are non-deterministic.
+
+.. _expectedEvent_connectionCheckOutFailedEvent:
+
+- ``connectionCheckOutFailedEvent``: Optional object. Assertions for one or more
+  `ConnectionCheckOutFailedEvent
+  <../connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#events>`__
+  fields.
+
+  The structure of this object is as follows:
+
+  - ``reason``: Optional string. Test runners MUST assert that the reason in the
+    published event matches this value. Valid values for this field are defined
+    in the CMAP spec.
+
+.. _expectedEvent_connectionCheckedOutEvent:
+
+- ``connectionCheckedOutEvent``: Optional object. If present, this object
+  MUST be an empty document as all fields in this event are non-deterministic.
+
+.. _expectedEvent_connectionCheckedInEvent:
+
+- ``connectionCheckedInEvent``: Optional object. If present, this object
+  MUST be an empty document as all fields in this event are non-deterministic.
+
+
+hasServerId
+```````````
+
+This field is an optional boolean that specifies whether or not the ``serverId``
+field of an event is set. If true, test runners MUST assert that the field is
+set and is a non-empty BSON ObjectId (i.e. all values of the ObjectId are not 0).
```

Comment:
I only made this distinction because a reasonable Go API could be something like

```
type CommandStartedEvent struct {
    ServerID primitive.ObjectID
}
```
In this case, an "unset" server ID would actually be an array of 12 0's. For languages that do something similar, I wanted to make sure they don't consider such an ID to be a set value.

Updated to "all bytes".

Reply:
OK. I had a feeling some implementation might be using an "empty" ObjectId as its unset value. That makes sense. I'm good with just changing this to refer to "bytes" then.